kvs-indexing/src/main/java/org/gbif/kvs/indexing/species/NameUsageMatchIndexer.java

```diff
@@ -105,34 +107,47 @@ private static void run(NameUsageMatchIndexingOptions options) {
                       new SaltedKeyGenerator(
                           storeConfiguration.getHBaseKVStoreConfiguration().getNumOfKeyBuckets());
 
-                  private transient NameMatchService nameMatchService;
+                  private transient ChecklistbankService checklistbankService;
 
                   private transient BiFunction<byte[], NameUsageMatch, Put> valueMutator;
 
                   @Setup
                   public void start() {
-                    nameMatchService = new NameMatchServiceSyncClient(nameMatchClientConfiguration);
+                    checklistbankService = new ChecklistbankServiceSyncClient(nameMatchClientConfiguration);
                     valueMutator =
                         NameUsageMatchKVStoreFactory.valueMutator(
                             Bytes.toBytes(storeConfiguration.getHBaseKVStoreConfiguration().getColumnFamily()),
                             Bytes.toBytes(storeConfiguration.getValueColumnQualifier()));
                   }
 
+                  /**
+                   * Retrieves the IUCN Red List category from the name usage match response.
+                   */
+                  private ThreatStatus getIucnRedListCategory(NameUsageMatch nameUsageMatch) {
+                    Integer usageKey = Objects.nonNull(nameUsageMatch.getAcceptedUsage()) ? nameUsageMatch.getAcceptedUsage().getKey() : nameUsageMatch.getUsage().getKey();
+                    Map<String,String> category = checklistbankService.getIucnRedListCategory(usageKey);
+                    if (Objects.nonNull(category)) {
+                      return ThreatStatus.valueOf(category.get("category"));
```

Comment:
Same as above, thinking if we should handle cases where the category is not present in the enum